src/jit/codegencommon.cpp

```diff
@@ -4282,7 +4282,14 @@ void CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg, bool* pXtraRegClobbere
 #endif // defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
         {
             // Bingo - add it to our table
-            regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            if (!compiler->opts.compUseSoftFP)
+            {
+                regArgNum = genMapRegNumToRegArgNum(varDsc->lvArgReg, regType);
+            }
+            else
+            {
+                regArgNum = genMapIntRegNumToRegArgNum(varDsc->lvArgReg);
+            }
 
```

Comment:
Right. For SoftFP, the function has no incoming floating-point register arguments, since all FP arguments are passed as integer arguments (as if using the varargs ABI). So it seems that the mask of incoming FP argument registers (namely, `regState->rsCalleeRegArgMaskLiveIn`) should be empty.